src/utils/toc.ts

```diff
@@ -0,0 +1,41 @@
+import slugger from 'github-slugger';
+
+interface ToCItem {
+  text: string;
+  id: string;
+  level: string;
+  subItems: ToCItem[];
+}
+
+export type TableOfContents = ToCItem[];
+
+export function getTableOfContents(mdxContent: string) {
+  const regexp = new RegExp(/^(### |## )(.*)\n/, 'gm');
+  const headings = [...mdxContent.matchAll(regexp)];
+  let tableOfContents: TableOfContents = [];
+
+  if (headings.length) {
+    let lastParentHeading: ToCItem;
+    headings.forEach((heading) => {
+      const headingText = heading[2].trim();
+      const headingType = heading[1].trim() === '##' ? 'h2' : 'h3';
+      const headingLink = slugger.slug(headingText, false);
+
+      const headingItem: ToCItem = {
+        text: headingText,
+        id: headingLink,
+        level: headingType,
+        subItems: [],
+      };
+
+      if (headingType === 'h2') {
+        tableOfContents.push(headingItem);
+        lastParentHeading = headingItem;
+      } else {
+        lastParentHeading.subItems.push(headingItem);
+      }
```

Comment:
We should do this recursively to construct the ToC from `h1` to `h6`.

Reply:
implemented a logic which appends it to the correct upper heading, inspired by rehype-toc

also added a test to ensure it's working as expected